Source/JavaScriptCore/runtime/ProxyObject.cpp

```diff
@@ -82,6 +86,64 @@ void ProxyObject::finishCreation(VM& vm, JSGlobalObject* globalObject, JSValue t
     internalField(Field::Handler).set(vm, this, handler);
 }
 
+JSObject* ProxyObject::getHandlerTrap(JSGlobalObject* globalObject, JSObject* handler, CallData& callData, const Identifier& ident, HandlerTrap trap)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    auto ensureIsCallable = [&](JSValue value) -> JSObject* {
+        if (value.isUndefinedOrNull())
+            return nullptr;
+
+        callData = JSC::getCallData(value);
+        if (callData.type == CallData::Type::None) {
+            throwTypeError(globalObject, scope, makeString("'", String(ident.impl()), "' property of a Proxy's handler should be callable"));
+            return nullptr;
+        }
+
+        return asObject(value);
+    };
+
+    JSObject* objectPrototype = globalObject->objectPrototype();
```

Comment:
This `globalObject` is lexical one, but we should check against ProxyObject's JSGlobalObject.

Reply:
YES thank you that was a real bug right there, and even switching to ProxyObject's wouldn't be a 100% correct fix. We can't avoid calling `handler->getPrototypeDirect()`.